### PR TITLE
Document the boost multiprecision shift bug in RuleList

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by chriseth** [comment]
Why is it only relevant here? We should force the boost version so that the bug is fixed.

File: libdevcore/CommonData.h [hunk 1]
```diff
@@ -140,6 +140,12 @@ inline bytes toCompactBigEndian(uint8_t _val, unsigned _min = 0)
 	return (_min || _val) ? bytes{ _val } : bytes{};
 }
 
+/// Workarounds shift left bug in boost <1.65.1.
```
Comment (Marenz) [inline libdevcore/CommonData.h:143]: Maybe link the issue or at least mention the issue number?

**Comment by axic** [comment]
> Why is it only relevant here? We should force the boost version so that the bug is fixed.

It only happens if the shift amount has multiple bits set and at least one of them is shifted out. The rest of the cases in `RuleList` only shift `u256(1)` so those should be fine. Not sure about the rest of the codebase.

This PR is only about documentation/explanation. I hope we can have another PR which force 1.65.1.

File: libdevcore/CommonData.h (hunk 1 above)
Comment (axic) [reply libdevcore/CommonData.h:143]: https://github.com/boostorg/multiprecision/issues/31

**Comment by chriseth** [comment]
I disagree, it is not just about documentation. It implements a workaround. I would prefer to use the proper fix instead of a workaround.

**Comment by axic** [comment]
It doesn't implement the workaround. The workaround is there already, this pulls it out into a helper.

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
Ah, I see, sorry!

File: libdevcore/CommonData.h (hunk 1 above)
Comment (Marenz) [reply libdevcore/CommonData.h:143]: I meant in the commend ^^ I mean that's where people will see it first and will be curious right?

**Pull request merged**